code/modules/spells/spell_types/spacetime_distortion.dm

```diff
@@ -0,0 +1,107 @@
+/obj/effect/proc_holder/spell/spacetime_dist
+	name = "Spacetime Distortion"
+	desc = "Entangle the strings of spacetime to deny easy movement around you. The strings vibrate..."
+	charge_max = 300
+	var/duration = 150
+	range = 7
+	var/list/effects
+	var/ready = TRUE
+	centcom_cancast = FALSE
+	sound = "sound/effects/magic.ogg"
+
+/obj/effect/proc_holder/spell/spacetime_dist/can_cast(mob/user = usr)
+	if(ready)
+		return ..()
+	return FALSE
+
+/obj/effect/proc_holder/spell/spacetime_dist/choose_targets(mob/user = usr)
+	var/list/turfs = spiral_range_turfs(range, user)
+	if(!turfs.len)
+		revert_cast()
+		return
+
+	ready = FALSE
+	var/list/turf_steps = list()
+	var/length = round(turfs.len * 0.5)
+	for(var/i in 1 to length)
+		turf_steps[pick_n_take(turfs)] = pick_n_take(turfs)
+	if(turfs.len > 0)
+		var/turf/loner = pick(turfs)
+		turf_steps[loner] = pick(Z_TURFS(user.z))
```

Comment:
There's always either an even number of tiles or an even number of tiles + 1.

Before the for loop I set the number of iterations to the number of tiles in the 'turfs' list / 2, then floored, so it will always leave out one tile but match all the other tiles in the loop. The tile left out is the loner tile handled in the if that follows.